### DISCUSSION TITLE
Click on specifi Part of the 3D model

### DISCUSSION THREAD

**Discussion by mohamed00736** [sceneview/sceneform-android#204] [Q&A]
hello , i am trying to click on specifi part of the 3D model and do somthing like showing a toast after clicking . THANX

**Comment by ThomasGorisse** [comment]
Hi,

Use this:
https://github.com/SceneView/sceneform-android/blob/4bdb11418b900de0700e27d839ff9bc629f0e0dd/core/src/main/java/com/google/ar/sceneform/Scene.java#L138-L149

`HitTestResult` contains the touched node, the distance and position of the peek.

**Reply by weijiaqi** [reply]
I tried it, but it doesn't work. I have two cartoon characters in a glb file. I can't get the different returns when I click on the two people. Their results are always the same.

**Comment by ThomasGorisse** [comment]
You will soon have access to the glTF picked sub-entity thanks to the new Filament Pick API but that will be only available on SceneView (not Sceneform)
